pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -349,6 +349,12 @@ type FelixConfigurationSpec struct {
 	// BPFKubeProxyEndpointSlicesEnabled in BPF mode, controls whether Felix's
 	// embedded kube-proxy accepts EndpointSlices or not.
 	BPFKubeProxyEndpointSlicesEnabled *bool `json:"bpfKubeProxyEndpointSlicesEnabled,omitempty" validate:"omitempty"`
+	// BPFPSNATPorts sets the range from which we randomly pick a port if there is a source port
+	// collision. This should be within the ephemeral range as defined by RFC 6056 (1024–65535) and
+	// preferably outside the  ephemeral ranges used by common operating systems. Linux uses
+	// 32768–60999, while others mostly use the IANA defined range 49152–65535. It is not necessarily
+	// a problem if this range overlaps with the operating systems. [Default: 20000:20999]
+	BPFPSNATPorts *numorstring.Port `json:"bpfPSNATPorts,omitempty"`
```

Comment:
Where does validation live now?  It definitely feels like we need validation for this field, to check that it is (a) numeric, as opposed to named and (b) a range, as opposed to a single port.

Reply:
Part of the validation is done by the type here (btw. disallows a single item range like x:x - that is imo not an issue given the use) and the rest is validated by the `portrange` type here  that in fact allows x:x and disallows named port. So I think all is covered. https://github.com/projectcalico/felix/blob/5b81ff13307d62909668a986bf8c91958e60a973/config/config_params.go#L179

Reply:
I agree that we have useful validation in Felix, but for many FelixConfig fields we have validation in libcalico-go as well.

E.g. the `KubeNodePortRanges` field has a validate tag: `validate:"omitempty,dive"`, which means it hits validation here: https://github.com/projectcalico/libcalico-go/blob/master/lib/validator/v3/validator.go#L645

Do you think the `BPFPSNATPorts` is already hitting that same validation as well?  I guess it might need a similar `validate` tag.  Also it would be good to add some tests in lib/validator/v3/validator_test.go, to make sure.

Reply:
`dive` isnt needed here because it is not a set of ranges, but a single range (unlike KubeNodePortRanges) so I think we are good on making sure that it is a valid range of ports